### ISSUE TITLE
[rez-memcache --flush] IndexError: list index out of range

### ISSUE THREAD

**Issue by ruzette** [AcademySoftwareFoundation/rez#1563]
Running `rez-memcache --flush` results in an `IndexError: list index out of range`.

**Environment**
* OS linux / Centos 7.9.2009 / Rocky Linux 9.2 
* Rez version 2.113.0
* Rez python version (output of "rez-python --version") python-3.6 (Centos 7.9) python-3.11 (Rocky 9.2)

**To Reproduce**
1. Typing `rez-memcache --flush` 

**Expected behavior**
No errors 

**Actual behavior**
```
$ /opt/rez/bin/rez/rez-memcache --flush
Traceback (most recent call last):
  File "/opt/rez/bin/rez/rez-memcache", line 8, in <module>
    sys.exit(run_rez_memcache())
             ^^^^^^^^^^^^^^^^^^
  File "/opt/rez/lib64/python3.11/site-packages/rez/cli/_entry_points.py", line 176, in run_rez_memcache
    return run("memcache")
           ^^^^^^^^^^^^^^^
  File "/opt/rez/lib64/python3.11/site-packages/rez/cli/_main.py", line 191, in run
    returncode = run_cmd()
                 ^^^^^^^^^
  File "/opt/rez/lib64/python3.11/site-packages/rez/cli/_main.py", line 183, in run_cmd
    return func(opts, opts.parser, extra_arg_groups)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/rez/lib64/python3.11/site-packages/rez/cli/memcache.py", line 99, in command
    memcache_client.flush(hard=True)
  File "/opt/rez/lib64/python3.11/site-packages/rez/utils/memcached.py", line 148, in flush
    self.reset_stats()
  File "/opt/rez/lib64/python3.11/site-packages/rez/utils/memcached.py", line 166, in reset_stats
    self._get_stats("reset")
  File "/opt/rez/lib64/python3.11/site-packages/rez/utils/memcached.py", line 189, in _get_stats
    return self.client.get_stats(stat_args=stat_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/rez/lib64/python3.11/site-packages/rez/vendor/memcache/memcache.py", line 331, in get_stats
    serverData[stats[1]] = stats[2]
                           ~~~~~^^^
IndexError: list index out of range
```

**Related Issues/PRs**
* #1241 (delete this section if N/A)

**Regression**
If applicable, state the last known rez version where the bug did not occur (delete this section if N/A).

**Comment by JeanChristopheMorinPerso** [comment]
It is failing at https://github.com/AcademySoftwareFoundation/rez/blob/371644d19de24d720cc0a4b14f6c17eeb61c55e1/src/rez/vendor/memcache/memcache.py#L330.

We send a `stats reset` command, which gets sent at https://github.com/AcademySoftwareFoundation/rez/blob/371644d19de24d720cc0a4b14f6c17eeb61c55e1/src/rez/vendor/memcache/memcache.py#L322.

For some reasons, the memcache library we use expects the response to contains a space delimited list of two items.

But memcache returns `RESULT`. Here is an example with telnet (memcache is a really simple protocol, so it's possible to use telnet to send requests):
```
[jcmorin@arch01 ~]$ telnet localhost 11211
Trying ::1...
Connected to localhost.
Escape character is '^]'.
stats reset
RESET
```

**Comment by JeanChristopheMorinPerso** [comment]
I know this worked before since I used `--flush` multiple times in the past. I was thinking the issue might have been with newer python versions, but looking at the code and the raw requests and responses, it doesn't seem python related... I also tried with memcached 1.6, 1.5 and 1.4. Same result...

**Comment by JeanChristopheMorinPerso** [comment]
Ok, I think I understand. Our vendored `python-memcache` module was updated from 1.53 to 1.58 in rez 2.44.0. The thing is that our vendored version had a custom patch. If I compare `src/rez/vendor/memcache/memcache.py` in rez 2.43.0 with the original from `python-memcache` 1.53, I get the following diff:

```diff
[jcmorin@arch01 rez]$ git diff src/rez/vendor/memcache/memcache.py /tmp/python-memcached/memcache.py 
diff --git a/src/rez/vendor/memcache/memcache.py b/tmp/python-memcached/memcache.py
index 7907eda9..9e511bd6 100644
--- a/src/rez/vendor/memcache/memcache.py
+++ b/tmp/python-memcached/memcache.py
@@ -282,8 +282,7 @@ class Client(local):
             readline = s.readline
             while 1:
                 line = readline()
-                if not line or line.strip() in ('END', 'RESET'):
-                    break
+                if not line or line.strip() == 'END': break
                 stats = line.split(' ', 2)
                 serverData[stats[1]] = stats[2]
```

This custom patched was accidentally dropped during the last upgrade 4 years ago. See https://github.com/AcademySoftwareFoundation/rez/commit/e65bb2145ed543ffaa9fdb1639055777b24bffcd#diff-f479f6b43fe6e341e5fdbd36b2ac6c5b21809957aec7b23553b782aa5829327fL285.

And the reason why it was working for me before is that I stopped being a rez user approximately 4 years ago (actually less than that) and the last last studio I worked at didn't have the latest version at the time.

The custom patch was introduced in https://github.com/AcademySoftwareFoundation/rez/commit/b3a8e344ef8dcc56337027d6739026b60141b9aa#diff-f479f6b43fe6e341e5fdbd36b2ac6c5b21809957aec7b23553b782aa5829327f (in 2014) right in the middle of the rewrite that happened between rez 1 and 2.
